application/client/src/app/ui/views/toolbar/chart/chart/state.ts

```diff
@@ -0,0 +1,377 @@
+            };
+            this._filters.update();
+        }
+        this._loading = false;
+        this._parent.detectChanges();
+    }
+
+    private _extractData(range: IRange | undefined) {
+        const filters: StoredEntity<FilterRequest>[] = this._session.search
+            .store()
+            .filters()
+            .get()
+            .filter((f) => f.definition.active);
+        this._datasets = [];
+        this._labels = new Array(this._map.length).fill('');
+        const map = range ? this._map.slice(range.begin, range.end) : this._map;
+        filters.forEach((filter: StoredEntity<FilterRequest>) => {
+            this._datasets.push({
+                barPercentage: 1,
+                categoryPercentage: 1,
+                label: filter.definition.uuid,
+                backgroundColor: filter.definition.colors.background,
+                data: [],
+            });
+        });
+        map.forEach((value: number[][], line: number) => {
+            value.forEach((matches: number[]) => {
+                const index: number = matches[0];
+                const data: number = matches[1];
+                const filter: StoredEntity<FilterRequest> = filters[index];
+                if (index === undefined || data === undefined || filter === undefined) {
+                    return;
+                }
+                this._datasets[index].data[line] = data;
+                this._datasets[index].backgroundColor = filter.definition.colors.background;
+            });
+        });
+    }
+
+    private _createChart() {
+        this._filters = new Chart(`view-chart-canvas-filters-${this._session.uuid()}`, {
+            type: 'bar',
+            data: {
+                labels: this._labels,
+                datasets: this._datasets,
+            },
+            options: {
+                plugins: {
+                    title: {
+                        display: false,
+                    },
+                    legend: {
+                        display: false,
+                    },
+                    tooltip: {
+                        enabled: false,
+                    },
+                },
+                animation: false,
+                responsive: true,
+                maintainAspectRatio: false,
+                scales: {
+                    y: {
+                        display: false,
+                        stacked: true,
+                        beginAtZero: true,
+                    },
+                    x: {
+                        stacked: true,
+                        display: false,
+                    },
+                },
+            },
+        });
+    }
+
+    private _fetch(width: number): Promise<void> {
+        return this._session.search
+            .getScaledMap(width)
+            .then((map) => {
+                this._map = map;
+                this._labels = new Array(this._map.length).fill('');
```

Comment:
may be I've missed something, but looks like `this._labels` has empty strings only. Do we need to keep it as property in this case? Or it's performance question (to avoid recreating array each time)?

Reply:
Yes, it's for performance only. We need labels or the data won't be drawn in the chart. Because we won't show the axis anyway, it's just filled with empty strings.
Maybe it makes sense to recreate the array every time anew since it's rare that the array is reused.